Front-End/challenge2/script.js

```diff
@@ -5,3 +5,104 @@ TO-DO:
 - The calculator should be completely functional
 
 */
+
+"use strict"
+
+const wrapper = document.querySelector('.wrapper')
+
+let numberStr = ''
+let viewString = ''
+let operations = ['+', '-', 'X', '/']
+let stack = []
+
+const updateValue = (value) => {
+    document.getElementById('display').textContent = value;
+}
+
+const evaluateStack = (arrStack) => {
+
+    //Convertimos a Notacion Posfija
+    if (arrStack.length == 0)
+        return 0;
+
+    let auxStack = []
+    let lastOp = '$';
+    for (let i = 0; i < arrStack.length; i++) {
+        if (operations.includes(arrStack[i])) {
+            lastOp = arrStack[i];
+        } else {
+            auxStack.push(arrStack[i])
+            if (lastOp != '$') {
+                auxStack.push(lastOp)
+            }
+        }
+    }
+    console.log(auxStack)
+    
+
+
+    let mainStack = []
+    let last;
+    mainStack.push(parseFloat(auxStack[0]));
+
+    for (let i = 1; i < auxStack.length; i++) {
+        let current = auxStack[i]
+        if (!operations.includes(current)) {
+            mainStack.push(current)
+        } else {
+            switch (current) {
+                case '+':
+                    last = mainStack.pop()
+                    mainStack[mainStack.length - 1] += parseFloat(last);
+                    break;
+                case '-':
+                    last = mainStack.pop()
+                    mainStack[mainStack.length - 1] -= parseFloat(last);
+                    break;
+                case '/':
+                    last = mainStack.pop()
+                    mainStack[mainStack.length - 1] /= parseFloat(last);
+                    break;
+                case 'X':
+                    last = mainStack.pop()
+                    mainStack[mainStack.length - 1] *= parseFloat(last);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+    return mainStack[0]
+}
+
+
+
+wrapper.addEventListener('click', (e1) => {
+    let val = e1.target.textContent
```

Comment:
what is `val`? please use more descriptive names, also why is a let, please try to review all your code and make sure your are using let and const properly

Reply:
Fixed